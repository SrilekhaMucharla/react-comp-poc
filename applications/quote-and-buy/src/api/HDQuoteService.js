import { getProxiedServiceUrl } from 'gw-portals-url-js';
import { BaseTransportService } from 'gw-portals-transport-js';

function processSubmission(method, data) {
    return BaseTransportService.send(
        getProxiedServiceUrl('quote'),
        {
            'Content-Type': 'application/json; charset=utf-8'
        },
        {
            jsonrpc: '2.0',
            method,
            params: [data]
        }
    );
}

function customizeSubmission(method, data) {
    return BaseTransportService.send(
        getProxiedServiceUrl('customquote'),
        {
            'Content-Type': 'application/json; charset=utf-8'
        },
        {
            jsonrpc: '2.0',
            method,
            params: [data]
        }
    );
}

export default class HDQuoteService {
    static createSubmission(data) {
        return processSubmission('create', data);
    }

    static createQuote(data) {
        return processSubmission('lwrSaveAndQuote', data);
    }

    static updateQuote(data) {
        return processSubmission('updateDraftSubmission', data);
    }

    static retrieveQuote(data) {
        return processSubmission('retrieve', data);
    }

    static retrievemulticarQuote(data) {
        return processSubmission('retrieveMultiProduct', data);
    }

    static retrieveSubmission(data) {
        return processSubmission('retrieveSubmission', data);
    }

    static customUpdateQuote(data) {
        return customizeSubmission('updateQuote', data);
    }

    static updateQuoteCoverages(data) {
        return customizeSubmission('updateQuoteCoverages', data);
    }

    static updateMarketingPreference(data) {
        return processSubmission('updateMarketingPreferences', data);
    }

    static updateMarketingPreferencesForMC(data) {
        return processSubmission('updateMarketingPreferencesForMC', data);
    }

    static sendEmailNotification(data) {
        return processSubmission('sendEmailNotification', data);
    }

    static customUpdateMultiQuote(data) {
        return customizeSubmission('updateMultiQuote', data);
    }

    static customUpdateMultiQuoteCoverages(data) {
        return customizeSubmission('updateMultiQuoteCoverages', data);
    }

    static updateMultiQuote(data) {
        return processSubmission('updateDraftMultiProduct', data);
    }

    static multiQuote(data) {
        return processSubmission('multiQuote', data);
    }

    static multiToSingleQuote(data) {
        return processSubmission('multiToSingleProduct', data);
    }

    static bindAndIssueService(data) {
        return processSubmission('bindAndIssue', data);
    }

    static updateSelectedVersion(data) {
        return processSubmission('updateSelectedVersion', data);
    }

    static updateProduct(data) {
        return processSubmission('updateProduct', data);
    }

    static singleToMultiProduct(data) {
        return processSubmission('singleToMultiProduct', data);
    }

    static applyDiscountOnMulticar(data) {
        return processSubmission('applyDiscountOnMulticar', data);
    }

    static updateSelectedVersionForMP(data) {
        return processSubmission('updateSelectedVersionForMP', data);
    }

    static mcBindAndIssue(data) {
        return processSubmission('mpBindAndIssue', data);
    }
}
