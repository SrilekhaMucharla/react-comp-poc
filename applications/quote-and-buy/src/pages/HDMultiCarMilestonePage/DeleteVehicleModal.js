import React from 'react';
import PropTypes from 'prop-types';
import { AnalyticsHDModal as HDModal } from '../../web-analytics';
import * as messages from './HDMultiCarMilestonePage.messages';
import { pageMetadataPropTypes } from '../../constant/propTypes';

const DeleteVehicleModal = ({
    onCancel, onConfirm, onClose, show, pageMetadata, hideClose
}) => (
    <HDModal
        webAnalyticsView={{ ...pageMetadata, page_section: messages.removeVehicle }}
        webAnalyticsEvent={{ event_action: messages.removeVehicle }}
        customStyle="rev-button-order"
        id="remove-vehicle-modal"
        show={show}
        headerText={messages.areYouSure}
        cancelLabel={messages.goBack}
        confirmLabel={messages.yes}
        onCancel={onCancel}
        onConfirm={onConfirm}
        onClose={onClose}
        hideClose={hideClose}
    >
        <p className="mb-5">{messages.removedCar}</p>
    </HDModal>
);

DeleteVehicleModal.propTypes = {
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
    onClose: PropTypes.func.isRequired,
    show: PropTypes.bool,
    hideClose: PropTypes.bool,
    pageMetadata: PropTypes.shape(pageMetadataPropTypes).isRequired
};

DeleteVehicleModal.defaultProps = {
    show: false,
    hideClose: false
};

export default DeleteVehicleModal;
