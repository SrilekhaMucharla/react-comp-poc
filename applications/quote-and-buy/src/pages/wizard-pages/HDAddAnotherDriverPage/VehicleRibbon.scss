.vehicle-ribbon {
    position: sticky;
    z-index: 4;
    top: $header-height-mobile;
    @include orientation(landscape) {
        top: 0;
        transition: .2s top;
        @include media-breakpoint-down(md) {
            .vehicle-ribbon-hidden-mobile-landscape & {
                top: -40px;
            }
        }
    }
    @include media-breakpoint-up(lg) {
        top: $header-height;
    }

    .text {
        text-transform: uppercase;
        font-size: 13px;
        line-height: 17.6px;
        background: $yellow-color;
        color: $app-theme;
        font-family: $font-licence-plate;
    }
}

.vehicle-ribbon--cq {
    @include media-breakpoint-up(lg) {
        padding: 0 35px;
    }

    .text {
        font-size: 15px;
        line-height: unset;
    }
}