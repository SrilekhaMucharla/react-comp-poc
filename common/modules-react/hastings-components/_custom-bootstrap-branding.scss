// Do nothing here

//TODO - move these definitions

///* import only the necessary Bootstrap files */
//@import "../../../node_modules/bootstrap/scss/functions";
//@import "../../../node_modules/bootstrap/scss/variables";
//@import "../../../node_modules/bootstrap/scss/mixins";
//
//@import "./custom-bootstrap-branding-variables";
//
///* import bootstrap to set changes */
//@import "../../../node_modules/bootstrap/scss/bootstrap";

//body {
//  font-family: $font-family-sans-serif;
//}


//.justify-content-md-center {
//  justify-content: center !important;
//}

//.btn-group-toggle {
//  display: unset;
//
//  > .btn:not(:last-child),
//  > .btn-group:not(:last-child) {
//    margin-right: 16px;
//  }
//
//  @include media-breakpoint-up(lg) {
//    > .btn:not(:last-child),
//    > .btn-group:not(:last-child) {
//      margin-right: 22px;
//    }
//  }
//
//  > .btn:not(:last-child):not(.dropdown-toggle),
//  > .btn-group:not(:last-child) > .btn {
//    @include border-radius(0.625rem);
//  }
//
//  > .btn:not(:first-child),
//  > .btn-group:not(:first-child) > .btn {
//    @include border-radius(0.625rem);
//  }
//
//
//  @include media-breakpoint-up(lg) {
//    > .btn {
//      width: 189px;
//      height: 60px;
//    }
//  }
//
//  > .btn {
//    display: inline-flex;
//    font-weight: 600;
//    font-size: 1.125rem;
//    margin-bottom: 30px;
//
//    > .justify-content-md-center {
//      align-items: center;
//      flex-direction: column;
//      display: flex;
//      width: 100%;
//    }
//  }
//
//  > .btn-primary:hover {
//    border-color: unset;
//  }
//}

//.horizontal-line {
//  width: 100%;
//  height: 1px;
//  border-radius: 6px;
//  background-color: $blue-transparent-30
//}