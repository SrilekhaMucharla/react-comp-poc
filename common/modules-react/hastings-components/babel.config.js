module.exports = require('./../../../common/build-tools/gw-build-config-babel');
