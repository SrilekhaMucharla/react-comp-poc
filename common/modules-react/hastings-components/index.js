import HDInteractiveCard from './components/HDInteractiveCard/HDInteractiveCard';
import HDInteractiveCardRefactor from './components/HDInteractiveCardRefactor/HDInteractiveCardRefactor';
import HDTextInput from './components/HDTextInput/HDTextInput';
import HDTextInputRefactor from './components/HDTextInputRefactor/HDTextInputRefactor';
import HDButton from './components/HDButton/HDButton';
import HDButtonRefactor from './components/HDButtonRefactor/HDButtonRefactor';
import HDInfoCard from './components/HDInfoCard/HDInfoCard';
import HDInfoCardRefactor from './components/HDInfoCardRefactor/HDInfoCardRefactor';
import HDQuoteInfo from './components/HDQuoteInfo/HDQuoteInfo';
import HDQuoteInfoRefactor from './components/HDQuoteInfoRefactor/HDQuoteInfoRefactor';
import HDQuoteInfoWarning from './components/HDQuoteInfoWarning/HDQuoteInfoWarning';
import HDLabel from './components/HDLabel/HDLabel';
import HDLabelRefactor from './components/HDLabelRefactor/HDLabelRefactor';
import HDStepper from './components/HDStepper/HDStepper';
import HDToggleButtonGroup from './components/HDToggleButtonGroup/HDToggleButtonGroup';
import HDToggleButtonGroupRefactor from './components/HDToggleButtonGroupRefactor/HDToggleButtonGroupRefactor';
import HDClickableCard from './components/HDClickableCard/HDClickableCard';
import HDDropdownList from './components/HDDropdownList/HDDropdownList';
import HDImageRadioButton from './components/HDImageRadioButton/HDImageRadioButton';
import HDButtonDashed from './components/HDButtonDashed/HDButtonDashed';
import HDDataCard from './components/HDDataCard/HDDataCard';
import HDForm from './components/HDForm/HDForm';
import HDWarningModal from './components/HDWarningModal/HDWarningModal';
import HDOverlayPopup from './components/HDOverlayPopup/HDOverlayPopup';
import HDModal from './components/HDModal/HDModal';
import HDAsyncSelect from './components/HDDropdownList/HDAsyncSelect';
import HDInformationToolTip from './components/HDInformationToolTip/HDInformationToolTip';
import HDAlert from './components/HDAlert/HDAlert';
import HDAlertRefactor from './components/HDAlertRefactor/HDAlertRefactor';
import HDDatePicker from './components/HDDatePicker/HDDatePicker';
import HDDatePickerRefactor from './components/HDDatePickerRefactor/HDDatePickerRefactor';
import HDTable from './components/HDTable/HDTable';
import HDAccordion from './components/HDAccordion/HDAccordion';
import HDAccordionRefactor from './components/HDAccordionRefactor/HDAccordionRefactor';
import HDQuoteTable from './components/HDQuoteTable/HDQuoteTable';
import HDQuoteDownload from './components/HDQuoteDownload/HDQuoteDownload';
import HDQuoteDownloadRefactor from './components/HDQuoteDownloadRefactor/HDQuoteDownloadRefactor';
import HDPlaceholderWithHeader from './components/HDPlaceholderWithHeader/HDPlaceholderWithHeader';
import HDContainer from './components/HDContainer/HDContainer';
import HDInfoBox from './components/HDInfoBox/HDInfoBox';
import HDTickListItem from './components/HDTickListItem/HDTickListItem';
import * as yup from './yup';
import HDSwitch from './components/HDSwitch/HDSwitch';
import HDBubble from './components/HDBubble/HDBubble';
import HDCompletedCardInfo from './components/HDCompletedCardInfo/HDCompletedCardInfo';
import HDRibbon from './components/HDRibbon/HDRibbon';
import HDFullscreenLoader from './components/HDFullscreenLoader/HDFullscreenLoader';
import HDSpinner from './components/HDSpinner/HDSpinner';
import HDLoader from './components/HDLoader/HDLoader';
import HDCustomerReviews from './components/HDCustomerReviews/HDCustomerReviews';
import HDCustomerReviewsWidget from './components/HDCustomerReviews/HDCustomerReviewsWidget';
import HDPaymentBreakdown from './components/HDPaymentBreakdown/HDPaymentBreakdown';
import HDToast from './components/HDToast/HDToast';
import HDCheckbox from './components/HDCheckbox/HDCheckbox';
import HDOverlay from './components/HDOverlay/HDOverlay';
import HDPriceTable from './components/HDPriceTable/HDPriceTable';
import HDVehicleSimpleDetails from './components/HDVehicleDetails/HDVehicleSimpleDetails';
import HDRadioButtonList from './components/HDRadioButtonList/HDRadioButtonList';
import HDCheckboxButtonList from './components/HDCheckboxButtonList/HDCheckboxButtonList';
import HDPolicySelect from './components/HDPolicySelect/HDPolicySelect';

export {
    HDTextInput,
    HDTextInputRefactor,
    HDButton,
    HDButtonRefactor,
    HDLabel,
    HDLabelRefactor,
    HDInfoCard,
    HDInfoCardRefactor,
    HDToggleButtonGroup,
    HDToggleButtonGroupRefactor,
    HDClickableCard,
    HDDropdownList,
    HDInteractiveCard,
    HDInteractiveCardRefactor,
    HDImageRadioButton,
    HDButtonDashed,
    HDStepper,
    HDDataCard,
    HDForm,
    HDWarningModal,
    HDModal,
    HDAsyncSelect,
    HDOverlayPopup,
    HDInformationToolTip,
    HDAlert,
    HDAlertRefactor,
    HDDatePicker,
    HDDatePickerRefactor,
    HDTable,
    HDAccordion,
    HDAccordionRefactor,
    HDQuoteTable,
    HDQuoteInfo,
    HDQuoteInfoRefactor,
    HDQuoteInfoWarning,
    HDQuoteDownload,
    HDQuoteDownloadRefactor,
    yup,
    HDPlaceholderWithHeader,
    HDTickListItem,
    HDContainer,
    HDBubble,
    HDInfoBox,
    HDSwitch,
    HDLoader,
    HDCompletedCardInfo,
    HDRibbon,
    HDFullscreenLoader,
    HDSpinner,
    HDCustomerReviews,
    HDCustomerReviewsWidget,
    HDPaymentBreakdown,
    HDToast,
    HDCheckbox,
    HDOverlay,
    HDPriceTable,
    HDVehicleSimpleDetails,
    HDRadioButtonList,
    HDCheckboxButtonList,
    HDPolicySelect
};
