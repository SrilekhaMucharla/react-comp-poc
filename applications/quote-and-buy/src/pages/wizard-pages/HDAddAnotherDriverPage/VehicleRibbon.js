/* eslint-disable react-hooks/rules-of-hooks */
import React, { useState } from 'react';
import { useDispatch, useSelector, connect } from 'react-redux';
import * as _ from 'lodash';
import PropTypes from 'prop-types';
import { HDRibbon } from 'hastings-components';
import { useHistory, useLocation } from 'react-router-dom';
import routes from '../../../routes/WizardRouter/RouteConst';
import twoWords from '../__helpers__/twoWords';
import * as messages from '../HDAddAnotherCarPage/HDAddAnotherCar.messages';
import DeleteVehicleModal from '../../HDMultiCarMilestonePage/DeleteVehicleModal';

import HDQuoteService from '../../../api/HDQuoteService';
import { getMultiToSingleParam } from '../../../common/utils';
import useFullscreenLoader from '../../Controls/Loader/useFullscreenLoader';
import {
    setNavigation as setNavigationAction,
    setErrorStatusCode,
    updateMultiQuote,
    resetCurrentPageIndex as resetCurrentPageIndexAction
} from '../../../redux-thunk/actions';
import { getDataForUpdateMultiQuoteAPICall } from '../../../common/submissionMappers';
import { trackAPICallSuccess, trackAPICallFail } from '../../../web-analytics/trackAPICall';
import { removeDataBasedOnPeriodStatus, removeOfferings } from '../../../common/submissionMappers/helpers';
import mcsubmission from '../../../routes/MCSubmissionVMInitial';
import {
    multiToSingle, multiToSingleFailed
} from '../../HDMCIntroPage/HDMCIntroPage.messages';

const VehicleRibbon = (props) => {
    const {
        vehicleAdded, fromMCDriverAllocation, setNavigation, resetCurrentPageIndex
    } = props;
    const dispatch = useDispatch();
    const location = useLocation();
    const history = useHistory();
    const [HDFullscreenLoader, showLoader, hideLoader] = useFullscreenLoader();
    const submissionVM = useSelector((state) => state.wizardState.data.submissionVM);
    const mcsubmissionVM = useSelector((state) => state.wizardState.data.mcsubmissionVM);
    const currentCarIndex = useSelector((state) => state.wizardState.app.currentPageIndex);
    const isEditQuoteJourney = useSelector((state) => state.wizardState.app.isEditQuoteJourney);
    const mcSingalQuoteEditObject = useSelector((state) => state.wizardState.app.MCSingalQuoteEditObject);
    let vehicles;
    if (vehicleAdded) {
        const vehicleIndex = useSelector((state) => state.wizardState.app.currentPageIndex);
        const checkMCAvailability = _.get(mcsubmissionVM, 'value.quotes', []);
        if (checkMCAvailability.length) {
            vehicles = useSelector(
                (state) => state.wizardState.data.mcsubmissionVM.quotes.children[vehicleIndex].lobData.privateCar.coverables.vehicles.value[0]
            );
        } else {
            vehicles = useSelector((state) => state.wizardState.data.submissionVM.lobData.privateCar.coverables.vehicles.value[0]);
        }
    } else {
        vehicles = useSelector((state) => state.wizardState.data.submissionVM.lobData.privateCar.coverables.vehicles.value[0]);
    }
    const vehicleReg = _.get(vehicles, 'registrationsNumber', null);
    const vehicleMake = _.get(vehicles, 'make', null);
    const vehicleModel = _.get(vehicles, 'model', null);
    const vehicleDetails = (vehicleReg) || `${vehicleMake} ${twoWords((vehicleModel) ? vehicleModel.toUpperCase() : null)}`;
    const QUOTES = 'value.quotes';

    const [deleteVehicle, setDeleteVehicle] = useState(null);
    const onDelete = () => setDeleteVehicle(true);
    const hideDeleteModal = () => setDeleteVehicle(false);

    const getMulticarSubmissionVMQuote = () => {
        return _.get(mcsubmissionVM, QUOTES, []);
    };

    let isChildCar = false;
    if (getMulticarSubmissionVMQuote().length) {
        isChildCar = true;
        if (location.pathname === routes.MC_POLICY_START_DATE || location.pathname === routes.MC_YOURQUOTE_PAGE) {
            const isParentCar = useSelector((state) => state.wizardState.data.mcsubmissionVM.quotes.children[currentCarIndex].isParentPolicy.value);
            isChildCar = !isParentCar;
        } else if (isEditQuoteJourney && submissionVM.value) {
            const isParentCar = submissionVM.value.isParentPolicy;
            isChildCar = !isParentCar;
        } else if (mcSingalQuoteEditObject && mcSingalQuoteEditObject.isParentPolicy) {
            const isParentCar = mcSingalQuoteEditObject.isParentPolicy;
            isChildCar = !isParentCar;
        }
    }

    // api trigger: update draft for multi product
    const triggerUpdateDraft = () => {
        return new Promise(async (resolve, reject) => {
            showLoader();
            await HDQuoteService.updateMultiQuote(getDataForUpdateMultiQuoteAPICall(mcsubmissionVM.value))
                .then(({ result }) => {
                    _.set(mcsubmissionVM, 'value', result);
                    trackAPICallSuccess('Update Draft MC Quote');
                    hideLoader();
                    resolve();
                }).catch((error) => {
                    dispatch(setErrorStatusCode(error.status));
                    trackAPICallFail('Update Draft MC Quote', 'Update Draft MC Quote Failed');
                    hideLoader();
                    reject();
                });
        });
    };

    // api trigger: multi to single product
    const triggerMultiToSingleAPI = () => {
        return new Promise(async (resolve, reject) => {
            showLoader();
            await HDQuoteService.multiToSingleQuote(getMultiToSingleParam(mcsubmissionVM))
                .then(({ result }) => {
                    hideLoader();
                    hideDeleteModal();
                    // update submissionVM, multi car initial state
                    _.set(submissionVM, 'value', result);
                    resetCurrentPageIndex();
                    dispatch(setNavigation({
                        quoteID: (result && result.quoteID) || '',
                        sessionUUID: (result && result.sessionUUID) || '',
                        softSellJourney: false
                    }));
                    resolve();
                }).catch((error) => {
                    hideLoader();
                    dispatch(setErrorStatusCode(error.status));
                    trackAPICallFail(multiToSingle, multiToSingleFailed);
                    reject();
                });
        });
    };

    const removeSelectedVehicleFromSubmission = () => {
        let mcQuotes = _.cloneDeep(_.get(mcsubmissionVM, QUOTES, []));
        mcQuotes = mcQuotes.filter((quoteObj) => {
            return quoteObj.quoteID !== _.get(submissionVM, 'value.quoteID', null);
        });
        _.set(mcsubmissionVM, QUOTES, _.cloneDeep(mcQuotes));
        resetCurrentPageIndex();
        return mcQuotes;
    };

    // navigate after each multi to single product api call
    const navigationAfterMultiToSingleServiceCall = () => {
        if (fromMCDriverAllocation) {
            const mcsubmissionInit = _.cloneDeep(mcsubmission);
            _.set(mcsubmissionVM, 'value', mcsubmissionInit);
            history.push(routes.MC_DRIVER_ALLOCATION);
        } else {
            dispatch(setNavigation({
                multiCarFlag: false
            }));
            const mcsubmissionInit = _.cloneDeep(mcsubmission);
            _.set(mcsubmissionVM, 'value', mcsubmissionInit);
            history.push(routes.ADD_ANOTHER_DRIVER);
        }
        trackAPICallSuccess(multiToSingle);
    };

    const onDeleteConfirm = () => {
        if (!deleteVehicle) return;
        dispatch(setNavigation({
            isEditQuoteJourney: false
        }));
        const mcsubmissionVMCloned = _.cloneDeep(mcsubmissionVM.value);
        if (location.pathname !== routes.MC_YOURQUOTE_PAGE && location.pathname !== routes.MC_POLICY_START_DATE) {
            removeSelectedVehicleFromSubmission();
            if (getMulticarSubmissionVMQuote().length > 1) {
                // mcsubmissioVMCloned: from this,  remove deleted quoteID
                dispatch(setNavigation({ fromMCDriverAllocation: false }));
                hideDeleteModal();
                // navigate to multicar milestone
                triggerUpdateDraft().then(() => {
                    history.push({
                        pathname: routes.MC_MILESTONE,
                        state: {
                            shouldNotUpdate: true
                        }
                    });
                });
            } else {
                triggerMultiToSingleAPI().then(() => {
                    navigationAfterMultiToSingleServiceCall();
                });
            }
        } else if (mcsubmissionVMCloned.quotes.length === 2) {
            // multi to single set submissionVM after successful response
            // navigate to mc driver allocation
            triggerMultiToSingleAPI().then(() => {
                navigationAfterMultiToSingleServiceCall();
            });
        } else {
            // length > 2
            // move quotes to draft state and then navigate the user to mc milestone page
            // we will do update here after removing deleted car from copy of mcsubmissioVM and then pass shoudlNotUpdate as true
            // currentCarIndex should not give error in mc policy start date page
            // will give error in MC YOURQUOTES PAGE (current car Index)
            showLoader();
            const mcsubmissionVMCopy = _.cloneDeep(mcsubmissionVM.value); // this should not have deleted car
            if (currentCarIndex > -1) _.pullAt(mcsubmissionVMCopy.quotes, currentCarIndex);
            // if periodStatus === qutoed, then remove bindData and offerings from payload
            // mcsubmissionVMCopy = getDataForUpdateMultiQuoteAPICall(mcsubmissionVMCopy);
            mcsubmissionVMCopy.quotes.map((paramObject) => {
                removeDataBasedOnPeriodStatus(paramObject, ['Quoted', 'Draft']); // Adding draft to fix issue regarding binddata send in update call
                removeOfferings(paramObject);
                return null;
            });
            HDQuoteService.updateMultiQuote(getDataForUpdateMultiQuoteAPICall(mcsubmissionVMCopy))
                .then(({ result }) => {
                    _.set(mcsubmissionVM, 'value', result);
                    resetCurrentPageIndex();
                    history.push({
                        pathname: routes.MC_MILESTONE,
                        state: {
                            shouldNotUpdate: true
                        }
                    });
                    hideLoader();
                    trackAPICallSuccess('Update Multi Quote');
                }).catch((error) => {
                    dispatch(setErrorStatusCode(error.status));
                    hideLoader();
                    trackAPICallFail('Update Multi Quote', 'Update Multi Quote Failed');
                });
        }
    };
    return (
        <>
            <HDRibbon
                text={vehicleDetails}
                actionText={isChildCar ? messages.removeCar : ''}
                onClick={onDelete}
                className="vehicle-ribbon" />
            {
                <DeleteVehicleModal
                    show={!!deleteVehicle}
                    onCancel={hideDeleteModal}
                    onConfirm={() => onDeleteConfirm()}
                    onClose={hideDeleteModal}
                    hideClose />
            }
            {HDFullscreenLoader}
        </>
    );
};

const mapStateToProps = (state) => {
    return {
        fromMCDriverAllocation: state.wizardState.app.fromMCDriverAllocation,
    };
};

const mapDispatchToProps = {
    setNavigation: setNavigationAction,
    updateMultiQuote,
    resetCurrentPageIndex: resetCurrentPageIndexAction,
};

VehicleRibbon.propTypes = {
    vehicleAdded: PropTypes.bool.isRequired,
    fromMCDriverAllocation: PropTypes.bool.isRequired,
    setNavigation: PropTypes.func.isRequired,
    resetCurrentPageIndex: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(VehicleRibbon);
