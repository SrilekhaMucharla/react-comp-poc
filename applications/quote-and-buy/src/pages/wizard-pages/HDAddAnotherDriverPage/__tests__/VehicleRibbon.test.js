import React from 'react';
import { mount } from 'enzyme';
import { HDRibbon } from 'hastings-components';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import VehicleRibbon from '../VehicleRibbon';
import createPortalRoot from '../../__helpers__/test/createPortalRoot';
import DeleteVehicleModal from '../../../HDMultiCarMilestonePage/DeleteVehicleModal';
import HDQuoteService from '../../../../api/HDQuoteService';
import multiCarSubmission from '../../../../routes/MCsubmissionMockTesting';

const middlewares = [];
// given
const mockedPush = jest.fn();
const isVehicleAdded = false;
const mockedRouterState = { isVehicleAdded };
const mockStore = configureStore(middlewares);
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockedPush
    }),
    useLocation: () => ({
        state: mockedRouterState
    })
}));
jest.mock('../../../../api/HDQuoteService');

const mockedVehicles = [
    { registrationsNumber: 'AV12BGE', make: '2012' },
    { registrationsNumber: 'S6TAY', make: '2013' }];
jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: () => mockedVehicles
}));

const mockedSetState = jest.fn();
jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: (init) => [init, mockedSetState]
}));

jest.mock('react-dom', () => ({
    ...jest.requireActual('react-dom'),
    createPortal: (element) => element
}));

const initializeStore = () => {
    const initialState = {
        wizardState: {
            app: {
                fromMCDriverAllocation: true
            },
        },
    };

    return mockStore(initialState);
};

async function initializeWrapper(store, props) {
    let wrapper;
    createPortalRoot();
    await act(async () => {
        wrapper = mount(
            <Provider store={store}>
                <VehicleRibbon {...props} />
            </Provider>
        );
    });

    HDQuoteService.updateMultiQuote.mockResolvedValue(multiCarSubmission);
    HDQuoteService.multiToSingleQuote.mockResolvedValue(multiCarSubmission);

    wrapper.update();
    return wrapper;
}

describe('RemoveVehicleRibbon', () => {
    // given

    it('should render correctly and match the snapshot', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store);
        expect(wrapper).toMatchSnapshot();
    });

    it('should contain HDRibbon', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store);
        expect(wrapper.find(HDRibbon)).toHaveLength(1);
    });

    it('should contain VehicleRibbon', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store);
        expect(wrapper.find(VehicleRibbon)).toHaveLength(1);
    });

    it('should call onDelete function', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store);
        wrapper.find('[className="vehicle-ribbon"]').simulate('click');
        expect(mockedSetState).toHaveBeenCalled();
    });

    it('should hide modal on confirm', async () => {
        const store = initializeStore();
        const wrapper = await initializeWrapper(store);
        wrapper.find('[className="vehicle-ribbon"]').simulate('click');
        // then
        await act(async () => wrapper.update());
        // then
        const modal = wrapper.find(DeleteVehicleModal);
        await act(async () => modal.props().onConfirm());
        await act(async () => wrapper.update());
        // then
        expect(modal.prop('show')).toBeFalsy();
    });
});
