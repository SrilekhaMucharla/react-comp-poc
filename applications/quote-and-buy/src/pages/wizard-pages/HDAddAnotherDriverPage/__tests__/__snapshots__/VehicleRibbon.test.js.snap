// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveVehicleRibbon should render correctly and match the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(VehicleRibbon)>
    <VehicleRibbon
      fromMCDriverAllocation={true}
      resetCurrentPageIndex={[Function]}
      setNavigation={[Function]}
      updateMultiQuote={[Function]}
    >
      <HDRibbon
        actionText=""
        className="vehicle-ribbon"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        text="null null"
      >
        <div
          className="ribbon vehicle-ribbon"
        >
          <div
            className="text"
          >
            null null
          </div>
        </div>
      </HDRibbon>
      <DeleteVehicleModal
        hideClose={true}
        onCancel={[Function]}
        onClose={[Function]}
        onConfirm={[Function]}
        show={false}
      >
        <ForwardRef
          cancelLabel="No, go back"
          confirmLabel="Yes, remove"
          customStyle="rev-button-order"
          headerText="Are you sure?"
          hideClose={true}
          id="remove-vehicle-modal"
          onBeforeOpen={null}
          onCancel={[Function]}
          onClose={[Function]}
          onConfirm={[Function]}
          show={false}
          webAnalyticsEvent={
            Object {
              "event_action": "Remove vehicle",
            }
          }
          webAnalyticsView={
            Object {
              "page_section": "Remove vehicle",
            }
          }
        >
          <ForwardRef
            cancelLabel="No, go back"
            confirmLabel="Yes, remove"
            customStyle="rev-button-order"
            headerText="Are you sure?"
            hideClose={true}
            id="remove-vehicle-modal"
            onBeforeOpen={[Function]}
            onBlur={null}
            onCancel={[Function]}
            onChange={null}
            onClick={null}
            onClose={[Function]}
            onConfirm={[Function]}
            onDelete={null}
            onEdit={null}
            onFocus={null}
            onSelect={null}
            show={false}
            webAnalyticsEvent={
              Object {
                "event_action": "Remove vehicle",
              }
            }
          >
            <HDModal
              cancelLabel="No, go back"
              className=""
              confirmLabel="Yes, remove"
              customStyle="rev-button-order"
              headerText="Are you sure?"
              hideCancelButton={false}
              hideClose={true}
              hideConfirmButton={false}
              hideFooter={false}
              id="remove-vehicle-modal"
              onBeforeOpen={[Function]}
              onCancel={[Function]}
              onClose={[Function]}
              onConfirm={[Function]}
              show={false}
            />
          </ForwardRef>
        </ForwardRef>
      </DeleteVehicleModal>
    </VehicleRibbon>
  </Connect(VehicleRibbon)>
</Provider>
`;
